packages/astro/src/core/build/plugins/plugin-manifest.ts

```diff
@@ -173,6 +174,19 @@ function buildManifest(
 		}
 	};
 
+	for (const route of opts.manifest.routes) {
+		if (!DEFAULT_COMPONENTS.find((component) => route.component === component)) {
+			continue;
+		}
+		routes.push({
+			file: '',
+			links: [],
+			scripts: [],
+			styles: [],
+			routeData: serializeRouteData(route, settings.config.trailingSlash),
+		});
+	}
```

Comment:
What are we doing here?

Reply:
Default routes are handled in a special way. We skip their processing during build but as a result, they are not part of `internals.pagesByKeys` so they didn't end up in the manifest. I'll add a comment explaining that

Reply:
It's worth leaving a comment then, because it's not understandable by looking only at the code